### DISCUSSION TITLE
Markdown template examples

### DISCUSSION THREAD

**Discussion by jayvdb** [lowlighter/metrics#610] [General]
The docs for the markdown templates are not very detailed, is a bit scary due to the matrix of unsupported plugins, and is a bit hard to follow being split across https://github.com/lowlighter/metrics/tree/master/source/plugins/core and https://github.com/lowlighter/metrics/tree/master/source/templates/markdown .  e.g. https://github.com/octocat/metrics referred to in the docs does not exist.

Not all of that is easy to fix quickly.  I think a good first step is to provide real examples.

e.g.
- https://github.com/hongaar/hongaar/blob/main/TEMPLATE.md for a simple example without embeds.
- https://github.com/Bangerok/Bangerok/blob/master/docs/template/README_TEMPLATE.md SVGs as assets and using `embed`
- https://github.com/kip93/kip93/blob/main/.github/workflows/readme.yml for a complex case where multiple svg are generated and then 'manually' embedded into the template

I found these using https://github.com/search?q=lowlighter+markdown+TEMPLATE.md&type=code

Could other community members link to other examples they know of that are doing interesting things?  I'd like to then update the relevant readme with practical examples.

**Comment by jayvdb** [comment]
The matrix of compatible plugins says "Base content" is supported, but there isn't a `base.ejs` in https://github.com/lowlighter/metrics/tree/master/source/templates/markdown/partials

**Comment by jayvdb** [comment]
It looks like `setup_community_templates` would be a useful mechanism for users to build their own markdown support for the various plugins, which would hopefully led to them being submitted upstream.  I am going to try this.

I quickly looked through https://github.com/search?l=YAML&p=1&q=setup_community_templates&type=Code , I can only see one use of `setup_community_templates`, which is the user activating [this minor change]( https://github.com/ezeholz/metrics/commit/94bdb23ee0d59ae25b0462299365551dbd37fb27) to image.svg .

**Comment by lowlighter** [comment]
Yeah I agree that the docs for markdown template is a bit light and can be improved

> Not all of that is easy to fix quickly. I think a good first step is to provide real examples.

Yes it's a good idea to add more examples. 
I'd prefer having them in an `markdown/examples` folder hosted in this repo rather than just linking users examples in case they choose to remove metrics from their repo (or even close it) so we don't have broken/irrevelant links

> The matrix of compatible plugins says "Base content" is supported, but there isn't a base.ejs 

It is probably hard-coded for `base` plugin. 
I'll patch the compatibility matrix to display either a ✅/☑️ with a legend to make it more understable that the markdown template actually supports all plugins

> It looks like setup_community_templates would be a useful mechanism for users to build their own markdown support for the various plugins, which would hopefully led to them being submitted upstream. I am going to try this.
>
> I quickly looked through https://github.com/search?l=YAML&p=1&q=setup_community_templates&type=Code , I can only see one use of setup_community_templates, which is the user activating this minor change to image.svg .

Basically `setup_community_templates` use case is to use official release while using a custom template from another user. 
This feature is probably not used a lot because I think users who made templates editions just use their forks directly (which makes sense).

**Reply by lowlighter** [reply]
The docs have been fully rewritten, hopefully it's now clearer how both markdown and community templates works 👍

**Comment by Andre601** [comment]
Not sure if mine is of any interest.
I mainly use the `{%- await embed(...) %}` stuff with the markdown template and configuration.

https://github.com/Andre601/Andre601/blob/master/TEMPLATE.md
